### PR TITLE
feat: use oatpp::DTO to parse img-input-connector APIData

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: src/imginputfileconn.h [hunk 1]
```diff
@@ -453,81 +455,92 @@ namespace dd
...
     {
+      auto params = ad.createSharedDTO<dd::DTO::ImgInputConnectorParameters>();
+
       // optional parameters.
-      if (ad.has("width"))
-        _width = ad.get("width").get<int>();
-      if (ad.has("height"))
-        _height = ad.get("height").get<int>();
-      if (ad.has("crop_width"))
+      if (params->width != nullptr)
```
Comment (beniz) [inline src/imginputfileconn.h:461]: Maybe we could write a short templated function that does the check against `nullptr` implicitely to simplify the code ?

File: src/imginputfileconn.h [hunk 2]
```diff
@@ -43,6 +43,8 @@
 #include "utils/apitools.h"
 #include <random>
 
+#include "http/dto/img_connector.hpp"
```
Comment (beniz) [inline src/imginputfileconn.h:46]: CI says you are missing this file !

File: src/imginputfileconn.h (hunk 1 above)
Comment (lganzzzo) [reply src/imginputfileconn.h:461]: you can do just:

```cpp
if (params->width) ... //<-- check for null
```

File: src/imginputfileconn.h [hunk 3]
```diff
@@ -453,81 +455,92 @@ namespace dd
...
-      if (ad.has("shuffle"))
-        _shuffle = ad.get("shuffle").get<bool>();
-      if (ad.has("seed"))
-        _seed = ad.get("seed").get<int>();
-      if (ad.has("test_split"))
-        _test_split = ad.get("test_split").get<double>();
-      if (ad.has("mean"))
+
+      if (params->bw != nullptr)
+        _bw = params->bw;
```
Comment (lganzzzo) [inline src/imginputfileconn.h:488]: `oatpp::Boolean` has the following implicit conversions:

```cpp
  {
    oatpp::Boolean ob = nullptr;
    bool b = ob;
    OATPP_ASSERT(b == false);
  }

  {
    oatpp::Boolean ob = false;
    bool b = ob;
    OATPP_ASSERT(b == false);
  }

  {
    oatpp::Boolean ob = true;
    bool b = ob;
    OATPP_ASSERT(b == true);
  }
```

**Review by beniz — APPROVED** [review]

**Review by fantes — APPROVED** [review]

**Pull request merged**